specification/2.0/Specification.adoc

```diff
@@ -146,91 +269,146 @@ The following are outside the scope of the initial design of glTF:
 
+Major version updates **MAY** be incompatible with previous versions.
 
-[[file-extensions-and-mime-types]]
-== File Extensions and MIME Types
 
-* `*.gltf` files use `model/gltf+json`
-* `*.bin` files use `application/octet-stream` or `application/gltf-buffer`
-  - When using MIME type `application/gltf-buffer`, the binary file extension may be `*.bin`, `*.glbin`, or `*.glbuf`.  See the https://www.iana.org/assignments/media-types/application/gltf-buffer[gltf-buffer registration].
-* Texture files use the official `image/*` type based on the specific image format. For compatibility with modern web browsers, the following image formats are supported: `image/jpeg`, `image/png`.
-+
-[NOTE]
-.Implementation Note
-====
-Implementations should use the image type pattern matching algorithm from the https://mimesniff.spec.whatwg.org/#matching-an-image-type-pattern[MIME Sniffing Standard] to detect PNG and JPEG images as file extensions may be unavailable in some contexts.
-====
+[[file-extensions-and-media-types]]
+== File Extensions and Media Types
+
+* JSON glTF files **SHOULD** use `.gltf` extension and <<gltf-json,`model/gltf+json`>> Media Type.
+* glTF files stored in GLB container **SHOULD** use `.glb` extension and <<gltf-binary,`model/gltf-binary`>> Media Type.
+* Files representing binary buffers **SHOULD** use either:
+** `.bin` file extension with <<octet-stream,`application/octet-stream`>> Media Type;
+** `.bin`, `.glbin`, or `.glbuf` file extensions with <<gltf-buffer,`application/gltf-buffer`>> Media Type.
+
+* <<png,PNG>> images **SHOULD** use `.png` file extension with <<image-png,`image/png`>> Media Type;
+** PNG images **SHOULD NOT** contain animations, non-square pixel ratios, or embedded ICC profiles. Such features if present **MUST** be ignored by client implementations.
 
+* <<jpeg,JPEG>> images **SHOULD** use `.jpeg` or `.jpg` file extensions with <<image-jpeg,`image/jpeg`>> Media Type
+** JPEG images **SHOULD** use <<jfif,JPEG File Interchange Format>>.
+** Client implementations **MAY** support JPEG images that use <<exif,Exchangeable image file format (Exif)>>.
 
 [[json-encoding]]
 == JSON Encoding
 
-To simplify client-side implementation, glTF has additional restrictions on JSON format and encoding.
+Although glTF Specification does not define any subset of the <<json,JSON>> format, implementations **SHOULD** be aware of its peculiar properties that could affect asset interoperability.
 
-1. JSON must use UTF-8 encoding without BOM.
+1. glTF JSON data **SHOULD** be written with UTF-8 encoding without BOM. This requirement is not applied when a glTF implementation does not control string encoding. glTF implementations **SHOULD** adhere to the <<json,RFC 8259>>, Section 8.1. with regards to treating BOM presence.
+
+2. ASCII characters stored in glTF JSON **SHOULD** be written unescaped.
```

Comment:
The earlier language seemed to say this referred to property names and enums but not values (e.g. URIs). Since some ASCII characters should be escaped in URIs, we may need to be more specific. "... written unescaped, with the exception of defined URI values."?

Reply:
The earlier language was causing more confusion than good and was not enforced anyway. IRIs are almost universally supported now, so only reserved characters have to be always percent-encoded (not the same as JSON escaping -- these are unrelated).

A simple example. Let's say a texture is called `🚀.png`. There are three valid options to encode it:

IRI within UTF-8 JSON (no escaping at all):
```json
{
  "uri": "🚀.png"
}
```

IRI within ASCII JSON (decoded by JSON parsers to the form above):
```json
{
  "uri": "\uD83D\uDE80.png"
}
```

URI within ASCII JSON (percent-encoding handled by URI parsers):
```json
{
  "uri": "%F0%9F%9A%80.png"
}
```

However when a filename contains _reserved_ characters (e.g. `#.png`), they always have to be percent-encoded.

Reply:
How about "... SHOULD be written without JSON escaping." to avoid ambiguity w/ URI escaping?